### PR TITLE
WINTERMUTE: Add more keyboard mappings, fix console warnings.

### PR DESCRIPTION
**Pull request by jammm**
Add mappings for numpad keys, tab, pause and backspace.
Suppressed unnecessary warnings for keys that are text-input.

### CODE REVIEW

**Comment by jammm** [comment]
@somaen Yes that's how it works. In white chamber you could press 7 without numlock and have the debug level selection mode, just like pressing home itself.

File: engines/wintermute/base/base_keyboard_state.cpp [hunk 1]
```diff
@@ -200,6 +200,12 @@ const char *BaseKeyboardState::scToString() {
 bool BaseKeyboardState::readKey(Common::Event *event) {
 	//_currentPrintable = (event->type == SDL_TEXTINPUT); // TODO
 	_currentCharCode = keyCodeToVKey(event);
+	// convert all lowercase keys to uppercase to make it easier for handling later on for consistency
+	if (_currentCharCode >= 'a' && _currentCharCode <= 'z' && (event->kbd.hasFlags(Common::KBD_SHIFT) || event->kbd.flags & Common::KBD_CAPS)) {
+		if (!(event->kbd.keycode >= Common::KEYCODE_F1 && event->kbd.keycode <= Common::KEYCODE_F12)) {
+			_currentCharCode -= 'a' - 'A';
```
Comment (somaen) [inline engines/wintermute/base/base_keyboard_state.cpp]: I think we have toupper available?

File: engines/wintermute/base/base_keyboard_state.cpp [hunk 2]
```diff
@@ -297,26 +308,53 @@ uint32 BaseKeyboardState::keyCodeToVKey(Common::Event *event) {
 		return 0;
 	}
 
+	// return ASCII value if key pressed is an alphanumeric key
+	// number keys pressed on numpad are handled in next block
+	if ((event->kbd.keycode >= 'a' && event->kbd.keycode <= 'z') || (event->kbd.keycode >= '0' && event->kbd.keycode <= '9')) {
```
Comment (somaen) [inline engines/wintermute/base/base_keyboard_state.cpp]: Can't Common::isAlnum be used for this?

File: engines/wintermute/base/base_keyboard_state.cpp [hunk 3]
```diff
@@ -297,26 +308,53 @@ uint32 BaseKeyboardState::keyCodeToVKey(Common::Event *event) {
...
+		return event->kbd.ascii;
+	}
+
+	// if NumLock is active, return ASCII for numpad keys
+	// keys pressed on numpad without NumLock are considered as normal keycodes, handled in the next block
+	if ((event->kbd.ascii >= '0' && event->kbd.ascii <= '9') && ((event->kbd.flags & Common::KBD_NUM) != 0)) {
```
Comment (somaen) [inline engines/wintermute/base/base_keyboard_state.cpp]: Similarly here.

**Comment by jammm** [comment]
All the requested changes have been been made.

**Pull request merged**